bin/beepmein

```diff
@@ -0,0 +1,281 @@
+        sleep 0.2
+    done
+}
+
+# $1 = how many times to blink, $2 = pause between each blink
+blink () {
+    # find primary connected or else 1st connected monitor
+    monitor="$(xrandr -q | grep -w "connected" | grep "primary" | cut -f1 -d ' ')"
+    if [ -z "$monitor" ]; then # no primary AND connected found, lets try 1st connected
+        monitor="$(xrandr -q | grep -w "connected" | cut -f1 -d ' ')"
+    fi 
+    [ -z "$monitor" ] && { >&2 echo "No monitors found error" ; exit 1; }
+    
+    # find current brightness of that monitor
+    now="$(xrandr --verbose | grep -A 15 "$monitor" | grep -i brightness | cut -f2 -d ' ')"
+    [ -z "$now" ] && { >&2 echo "Could not get current brightness error" ;  exit 1; }
+    
+    
+    [ -z "$1" ] && loops="1" || loops="$1"
+    for ((c=1; c<=loops; c++))
+    do
+        # blink up
+        for i in $(seq "$now" 0.2 3.0); do     
+            xrandr --output "$monitor" --brightness "$i"
+            sleep 0.01
+        done
+        
+        # blink down
+        for i in $(seq 3.0 -0.2 "$now"); do     
+            xrandr --output "$monitor" --brightness "$i"
+            sleep 0.01
+        done
+    sleep "$2"
+    xrandr --output "$monitor" --brightness "$now"
+    done
+}
+
+# alarm failsafe (by misko), this will be used if 'alarmSample' is not found.
+alarmfailsafe () {
+for ((i=1; i<=10; i++)); do
+      for ((c=1;c<=3;c++)); do
+          timeout -s HUP  0.2 speaker-test -t sine -f 3000  > /dev/null 2>&1
+          sleep 0.2
+       done
+       sleep 1
+done
+}
+
+# test alarm sample path (no test for reminder sample)
+testAlarmPath () {
+    [[ -f "$alarmSample" ]] || { >&2 superecho "Note: alarmSample $alarmSample not found, Internal fail-safe beep will be used instead."; }
+}
+
+# mpv or cvlc (by jr)
+playa () {
+    file="$1"
+    [[ -f $file ]] || { superecho "file not found" ; return 1; }
+    if command -v mpv >/dev/null 2>&1
+    then
+        mpv --no-resume-playback --no-video "$file"
+    elif command -v cvlc >/dev/null 2>&1
+    then
+        cvlc --no-video "$file" vlc://quit
+    else
+        superecho "no player"
+        return 1
+    fi
+}
+
+# test audio and notifications
+test () {
+    echo "notify test"
+    notify-send "notify test" || echo "failed"
+    echo
+    echo "fail-safe internal beep test used for reminders"
+    beep "3" || echo "failed"
+    echo
+    echo "mpv sample test used for alarms (either this or vlc must pass)"
+    mpv --no-resume-playback --no-video --length=10 "$alarmSample" || echo "mpv failed"
+    echo
+    echo "vlc sample test used for alarms"
+    cvlc --no-video --run-time=10 "$alarmSample" vlc://quit || echo "cvlc failed"
+    echo
+    echo "fail-safe internal beep test used for alarms"
+    alarmfailsafe || echo "failed"
+    echo
+}
+
+# validate 'at' time syntax by generating fake job
+validate () {
+tmpjob="$(at "$1" <<< "$(echo "test")" 2>&1 | grep ^job | cut -f2 -d ' ')"
```

Comment:
Use `<<<"test"`, no need to shell out

Reply:
I prefer to always name function arguments in bash, e.g.
```
somefunc() {
   local path=$1
}
```
it makes it easier to understand what each argument means.

Reply:
Ok, good one.